.devcontainer/README.md

```diff
@@ -0,0 +1,30 @@
+# Envoy Dev Container (experimental)
+
+This directory contains some experimental tools for Envoy Development in [VSCode Remote - Containers](https://code.visualstudio.com/docs/remote/containers).
+
+## How to use
+
+Open with VSCode with the Container extension installed. Follow the [official guide](https://code.visualstudio.com/docs/remote/containers) to open this
+repository directly from GitHub or from checked-out source tree.
+
+After opening, run the `Refresh Compilation Database` task to generate compilation database to navigate in source code. 
+This will run partial build of Envoy and may take a while depends on the machine performance. This task is needed to run everytime after
+changing BUILD files to get the changes reflected.
+
+## Advanced Usages
+
+### Using Remote Build Execution
+
+Write the following content to `devcontainer.env` and rebuild the container. The key will be persisted in the container's `~/.bazelrc`.
+
+```
+GCP_SERVICE_ACCOUNT_KEY=<base64 encoded service account key>
+BAZEL_REMOTE_INSTANCE=<RBE Instance>
+BAZEL_REMOTE_CACHE=grpcs://remotebuildexecution.googleapis.com
+BAZEL_BUILD_EXTRA_OPTIONS=--config=remote-ci --config=remote --jobs=<Number of jobs>
```

Comment:
Is it worth diving into download minimal vs. all here? I think that can make a huge difference if you are sitting behind a cable modem connection vs. same region GCP?

Reply:
added.